clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaStoreIngestionService.java

```diff
@@ -247,9 +243,7 @@ public KafkaStoreIngestionService(
         veniceConfigLoader.getVeniceClusterConfig().getClusterProperties().toProperties();
 
     veniceWriterProperties.put(PubSubConstants.PUBSUB_PRODUCER_USE_HIGH_THROUGHPUT_DEFAULTS, "true");
-    producerAdapterFactory = pubSubClientsFactory.getProducerAdapterFactory();
-    VeniceWriterFactory veniceWriterFactory =
-        new VeniceWriterFactory(veniceWriterProperties, producerAdapterFactory, metricsRepository);
+    VeniceWriterFactory veniceWriterFactory = new VeniceWriterFactory(veniceWriterProperties, metricsRepository);
```

Comment:
Let’s avoid creating factories at individual call sites. Instead, we should create the factory
once per server component and reuse it.